### PR TITLE
[SUCE-305] chinese lookup

### PR DESCRIPTION
**Pull request by gradam**

### CODE REVIEW

File: client/elements/lookups/sc-lzh2en.html [hunk 1]
```diff
@@ -1,20 +1,126 @@
...
+                    definition += this._lookupWord(snip);
+                    if(definition && i > length){
+                        length = i;
+                    }
+                }
+                return {html: definition, length: length};
+            }
+
+            _lookupWord(graph){
+                graph = graph.replace(/\u2060/, '');
```
Comment (TrebuhD) [inline client/elements/lookups/sc-lzh2en.html]: what's this symbol? Maybe extract it to a variable?

File: client/elements/lookups/sc-lzh2en.html [hunk 2]
```diff
@@ -1,20 +1,126 @@
...
+                                   href="${href}" target="_blank" class="lookup-link">${graph}</a>
+                              </span>
+                              <span class="meaning">
+                                ${this.fallbackDictData.dictionary[graph][0]}: ${this.fallbackDictData.dictionary[graph][1]} [modern chinese]
+                              </span>
+                              <hr>
+                            </div>`
+                }
+                return '';
+            }
```
Comment (TrebuhD) [inline client/elements/lookups/sc-lzh2en.html]: This function is too long. I'd extract the `if` blocks to separate functions.

File: client/elements/text/sc-simple-text.html [hunk 3]
```diff
@@ -301,76 +299,147 @@
...
+                          } else {
+                            str += `%spback%%spfrnt%${graph}%spback%%spfrnt%`;
+                          }
+                      }
+                      if(str.endsWith('%spfrnt%')){
+                          str = str.substring(0, str.length - '%spfrnt%'.length) + ' ';
+                      } else {
+                        str += '%spback% ';
+                      }
+                  }
```
Comment (TrebuhD) [inline client/elements/text/sc-simple-text.html]: this function is too long

File: client/elements/text/sc-simple-text.html [hunk 4]
```diff
@@ -301,76 +299,147 @@
...
+                            if(!nextElement.classList.contains('lookup_element')){
+                                i--;
+                                continue;
+                            }
+                            nextElement.classList.add('green-color');
+                        }
+                    }, 5);
+                }
+                tooltip.elementCount = lookupResult.length;
+              }
```
Comment (TrebuhD) [inline client/elements/text/sc-simple-text.html]: ditto. This should be at least 3 functions.

File: client/elements/text/sc-simple-text.html [hunk 5]
```diff
@@ -384,91 +453,153 @@
           }
 
           _removeTooltip(v, tooltip, states, e) {
-              let timeToWait = e.sourceCapabilities.firesTouchEvents ? 10 : 100;
+              let timeToWait = this._getTooltipDelayTime(e);
```
Comment (TrebuhD) [inline client/elements/text/sc-simple-text.html]: this can be inline

File: client/elements/text/sc-simple-text.html [hunk 6]
```diff
@@ -384,91 +453,153 @@
...
+                            break;
+                        }
+                        lastElement = nextElement;
+                        if(!nextElement.classList.contains('lookup_element')){
+                            i--;
+                            continue;
+                        }
+                        if(!nextElement.id){
+                            nextElement.classList.remove('green-color');
+                        }
```
Comment (TrebuhD) [inline client/elements/text/sc-simple-text.html]: the loop body could be a separate function

File: client/elements/text/sc-simple-text.html [hunk 7]
```diff
@@ -384,91 +453,153 @@
...
                   }
-              }, timeToWait);
+              }, 0);
           }
 
-          _addLookupTooltips(textContainer, self) {
-              self = self || this;
-              if (!self.isPaliLookupEnabled) {
-                  return;
+          _addLookupTooltips(textContainer, lang, that) {
```
Comment (TrebuhD) [inline client/elements/text/sc-simple-text.html]: ditto. Loop body needs to be a separate function.

File: client/elements/lookups/sc-lzh2en.html (hunk 1 above)
Comment (gradam) [reply client/elements/lookups/sc-lzh2en.html]: No idea what it is

File: client/elements/text/sc-simple-text.html (hunk 6 above)
Comment (gradam) [reply client/elements/text/sc-simple-text.html]: no need for that.

**Pull request merged**